src/components/Form/ManageUsers.jsx

```diff
@@ -24,118 +22,169 @@ import FormSection from './FormSection';
+  const [webId, setWebId] = useState('');
+  const [userEditingWebId, setUserEditingWebId] = useState(false);
   const { setUserList } = useContext(UserListContext);
 
   // Event handler for adding user from users list
   const handleAddUser = async (event) => {
     event.preventDefault();
-    dispatch({ type: 'SET_PROCESSING' });
     const userObject = {
-      givenName: event.target.addUserGivenName.value,
-      familyName: event.target.addUserFamilyName.value,
-      username: event.target.addUsername.value
+      givenName: userGivenName.value,
+      familyName: userFamilyName.value,
+      username,
+      webId
     };
 
-    if (!userObject.username) {
-      runNotification(`Operation failed. Reason: No username provided`, 5, state, dispatch);
-      setTimeout(() => {
-        dispatch({ type: 'CLEAR_PROCESSING' });
-      }, 3000);
-      return;
-    }
-
-    if (!userObject.givenName) {
-      runNotification(
-        `Operation failed. Reason: User's first/given name is not provided`,
-        5,
-        state,
-        dispatch
-      );
-      setTimeout(() => {
-        dispatch({ type: 'CLEAR_PROCESSING' });
-      }, 3000);
-      return;
-    }
-
-    if (!userObject.familyName) {
-      runNotification(
-        `Operation failed. Reason: User's last/family name is not provided`,
-        5,
-        state,
-        dispatch
-      );
-      setTimeout(() => {
-        dispatch({ type: 'CLEAR_PROCESSING' });
-      }, 3000);
-      return;
-    }
-
-    let listUsers = await addUserToPod(session, userObject);
-    listUsers = await getUserListActivity(session, listUsers);
+    const listUsers = await sendUserToSolid(userObject, dispatch, session, state);
 
     setUserList(listUsers);
-
-    runNotification(
-      `Adding user "${userObject.givenName} ${userObject.familyName}" to Solid...`,
-      5,
-      state,
-      dispatch
-    );
-
-    setTimeout(() => {
-      clearUserGivenName();
-      clearUserFamilyName();
-      clearUsername();
-      dispatch({ type: 'CLEAR_PROCESSING' });
-    }, 3000);
+    clearUserFamilyName();
+    clearUserGivenName();
+    setUsername('');
+    setWebId('');
   };
 
-  const formRowStyle = {
-    margin: '20px 0'
+  const wrappedSetUsername = (value) => {
+    setUsername(value);
+    if (userEditingWebId) {
+      return;
+    }
+    const renderedWebId = renderWebId(value);
+    setWebId(renderedWebId);
   };
 
-  /* eslint-disable jsx-a11y/label-has-associated-control */
   return (
     <FormSection
-      title="Manage Users"
+      title="Add New User"
       state={state}
       statusType="Status"
       defaultMessage="To be added..."
     >
       <form onSubmit={handleAddUser} style={formRowStyle} autoComplete="off">
-        <div>
-          <label htmlFor="add-user-given-name">First/given name: </label>
-          <input id="add-user-given-name" name="addUserGivenName" {...userGivenName} />{' '}
-        </div>
+        <TextField
```

Comment:
We could wrap these TextFields with FormControl components. This could also mitigate MUI errors for uncontrolled inputs. FormControl is helpful since we could then wrap TextFields with FormHelperText components. They'll look like this at the bottom below the TextField:

![Screen Shot 2023-05-20 at 17 04 54](https://github.com/codeforpdx/PASS/assets/14917816/93f920ca-c4e4-48f3-b1d6-0398c126a1dc)

You could replace that text on the bottom with something else.

Reply:
TextField actually already uses `FormControl` [components internally](https://mui.com/material-ui/api/text-field/#advanced-configuration). That helper text can be added in a TextField using the `helperText` parameter ([link to MUI demo's code](https://github.com/mui/material-ui/blob/0c58832b6e182f782e73b118ed9681055db3e249/docs/data/material/components/text-fields/FormPropsTextFields.tsx#LL55C33-L55C33). There may be some value in adding a `FormControl` into the `FormSection` just so we don't need to think about it, or adding helper text to the top of this whole form, but we can leave that for another PR.

Reply:
I also think those errors were being thrown because we weren't setting the `value` and `onChange` properties, not because of the FormControl

Reply:
Ah okay. I think I know what you're referring to.

I just read a bit more about it in the MUI documentation about controlled vs uncontrolled. https://mui.com/material-ui/react-text-field/#uncontrolled-vs-controlled